app/src/main/java/com/kamron/pogoiv/ScreenGrabber.java

```diff
@@ -144,4 +114,55 @@ private Bitmap getBitmap(ByteBuffer buffer, int pixelStride, int rowPadding) {
 
         return bmp;
     }
+
+    public @Nullable @ColorInt int[] grabPixels(Point[] points) {
+        Image image = null;
+        try {
+            //Note: imageReader shouldn't be null, but apparently sometimes is.
+            //Let's allow this to still happen.
+            image = imageReader.acquireLatestImage();
+        } catch (Exception exception) {
+            Timber.e("Error thrown in grabPixels() - acquireLatestImage()");
+            Timber.e(exception);
+        }
+
+        if (image == null) {
+            return null;
+        }
+
+        Rect imageBounds = new Rect(0, 0, image.getWidth(), image.getHeight());
+        @ColorInt int[] pixels = new int[points.length];
+
+        Image.Plane imgPlane = image.getPlanes()[0];
+        int pixelStride = imgPlane.getPixelStride();
+        int rowStride = imgPlane.getRowStride();
+        ByteBuffer buffer = imgPlane.getBuffer();
+
+        for (int i = 0; i < points.length; i++) {
+            Point p = points[i];
+            if (imageBounds.contains(p.x, p.y)) {
+                pixels[i] = getPixel(buffer, p, pixelStride, rowStride);
```

Comment:
Re mutable bitmaps: the implementation of `createBitmap` we use will often return a mutable bitmap—in fact, if I trust JavaDocs of other invoked `createBitmap` methods,  the `createBitmap` overload you mention will **always** return a mutable bitmap, unless the original bitmap is immutable and can be reused.